Dynamic-range-limit property example (#43822)
mdn/content: files/en-us/web/css/reference/properties/dynamic-range-limit/index.md

### Basic `dynamic-range-limit` usage

This example demonstrates the basic use of the `dynamic-range-limit` property and the difference between HDR and SDR images.

#### HTML

In the markup, we embed an HDR image using an {{htmlelement("img")}} element. We include a [`tabindex`](/en-US/docs/Web/HTML/Reference/Global_attributes/tabindex) value of `0` to make the image focusable via the keyboard.

```html
<img

>   src="https://mdn.github.io/shared-assets/images/examples/hi-res/lights.jpg"

Reviewer: link would need to be updated

Reviewer 2: ```suggestion
  src="https://mdn.github.io/shared-assets/images/examples/ultra-hdr.jpg"
```

>   alt="White strip lights lighting in a subway station"

Reviewer: or
```suggestion
  alt="A subway station platform with bright white overhead strip lights"
```

  tabindex="0" />
```

#### CSS

> We display the image as if it were and SDR image by setting the `dynamic-range-limit` property to `standard`, which means it won't be displayed any brighter than HDR reference white. We also set a {{cssxref("transition")}} property so that the `<img>` element's `dynamic-range-limit` value will transition over `0.6` seconds when its state changes.

Reviewer: ```suggestion
We constrain the image to SDR brightness by setting the `dynamic-range-limit` property to `standard`, which makes the image no brighter than HDR reference white. We also set a {{cssxref("transition")}} property so that the `<img>` element's `dynamic-range-limit` value transitions over `0.6` seconds when its state changes.
```

### Basic `dynamic-range-limit` usage

This example demonstrates the basic use of the `dynamic-range-limit` property and the difference between HDR and SDR images.

#### HTML

In the markup, we embed an HDR image using an {{htmlelement("img")}} element. We include a [`tabindex`](/en-US/docs/Web/HTML/Reference/Global_attributes/tabindex) value of `0` to make the image focusable via the keyboard.

```html
<img

>   src="hdr.jpg"

Reviewer 2: ```suggestion
  src="https://mdn.github.io/shared-assets/images/examples/ultra-hdr.jpg"
```
Once the [shared-asset PR](https://github.com/mdn/shared-assets/pull/95) is merged this ☝️ will start working.

>   src="hdr.jpg"

Author: ```suggestion
  src="https://mdn.github.io/shared-assets/images/examples/hi-res/lights.jpg"
```

}
```

```css hidden
img {
  max-height: 100vh;
}
@media not (dynamic-range: high) {
  body::before {
    content: "Your device may not display the image at full brightness.";
    background-color: wheat;
    display: block;
    text-align: center;
  }
}
@supports not (dynamic-range-limit: standard) {
  body::before {
    content: "Your browser doesn't support the dynamic-range-limit property.";
    background-color: wheat;
    display: block;
    text-align: center;
  }
}
```

#### Results

{{EmbedLiveSample("Examples", 300, 400)}}

> The image is HDR, but is displayed as muted by default. Hover or focus the image. Note how it transitions to vivid HDR colors in displays capable of displaying them.

Reviewer: ```suggestion
The image is ultra HDR but is constrained to SDR brightness by default. Hover over or focus the image. Note how it transitions to vivid HDR colors on capable displays.
```

Reviewer 2: > Note how it transitions to vivid HDR colors on capable displays.

We need to give examples of a few "capable" displays. There is browser support on the page but no device support information.
